clients/fides-js/src/lib/consent.tsx

```diff
@@ -0,0 +1,188 @@
+import { getConsentContext } from "./consent-context";
+
+/**
+ * Validate the config options
+ */
+const validateBannerOptions = (config: FidesConfig): boolean => {
+  // Check if options is an invalid type
+  if (config.options === undefined || typeof config.options !== "object") {
+    return false;
+  }
+  // todo- more validation here?
+
+  if (!config.options.privacyCenterUrl) {
+    debugLog(
+      config.options.debug,
+      "Invalid banner options: privacyCenterUrl is required!"
+    );
+    return false;
+  }
+
+  if (config.options.privacyCenterUrl) {
+    try {
+      // eslint-disable-next-line no-new
+      new URL(config.options.privacyCenterUrl);
+    } catch (e) {
+      debugLog(
+        config.options.debug,
+        "Invalid banner options: privacyCenterUrl is an invalid URL!",
+        config
+      );
+      return false;
+    }
+  }
+
+  return true;
+};
+
+/**
+ * Fetch the user's geolocation from an external API
+ */
+const getLocation = async (options: FidesOptions): Promise<UserGeolocation> => {
+  // assumes that isGeolocationEnabled is true
+  debugLog(options.debug, "Running getLocation...");
+  const { geolocationApiUrl } = options;
+
+  if (!geolocationApiUrl) {
+    debugLog(
+      options.debug,
+      "Location cannot be found due to no configured geoLocationApiUrl."
+    );
+    return {};
+  }
+
+  debugLog(
+    options.debug,
+    `Calling geolocation API: GET ${geolocationApiUrl}...`
+  );
+  const fetchOptions: RequestInit = {
+    mode: "cors",
+  };
+  const response = await fetch(geolocationApiUrl, fetchOptions);
+
+  if (!response.ok) {
+    debugLog(
+      options.debug,
+      "Error getting location from geolocation API, returning {}. Response:",
+      response
+    );
+    return {};
+  }
+
+  try {
+    const body = await response.json();
+    debugLog(
+      options.debug,
+      "Got location response from geolocation API, returning:",
+      body
+    );
+    return body;
+  } catch (e) {
+    debugLog(
+      options.debug,
+      "Error parsing response body from geolocation API, returning {}. Response:",
+      response
+    );
+    return {};
+  }
+};
+
+/**
+ * Initialize the Fides Consent Banner or Link, with optional extraOptions to override defaults.
+ *
+ * (see the type definition of ConsentBannerOptions for what options are available)
+ */
+export const initFidesConsent = async (config: FidesConfig): Promise<void> => {
+  debugLog(config.options.debug, "Initializing Fides consent...");
+
+  debugLog(
+    config.options.debug,
+    "Validating Fides consent banner options...",
+    config.options
+  );
+  if (!validateBannerOptions(config)) {
+    return Promise.reject(new Error("Invalid banner options"));
+  }
+
+  if (config.options.isDisabled) {
+    debugLog(
+      config.options.debug,
+      "Fides consent banner is disabled, skipping banner initialization!"
+    );
+    return Promise.resolve();
+  }
+
+  document.addEventListener("DOMContentLoaded", () => {
```

Comment:
We have to ensure the DOM is loaded, otherwise we'll see a preact error when we try to `render(<App config={config} />, document.body);`. I'll instead add a condition to test if DOM is loaded